### PR TITLE
FIX empty shopbuilder headers

### PR DESCRIPTION
**Pull request by cjohannsen**
- TwigStyleScriptTagFilter.php - scrip and style tags are not longer replaced via regex, because it causes PREG_BACKTRACK_LIMIT_ERROR

### All changes meet the following requirements
- [x] Changelog entry was added
- [x] Changes have been tested by the author
- [x] Changes have been tested by the reviewer

@plentymarkets/ceres-io

### CODE REVIEW

**Review by stentrop — COMMENTED** [review]
Preview Badge kommt doppelt raus

**Comment by cjohannsen** [comment]
> Preview Badge kommt doppelt raus

fixed

File: src/Extensions/TwigStyleScriptTagFilter.php [hunk 1]
```diff
@@ -103,35 +103,50 @@ public function filterTags($content, $containerName)
...
+
+                $element->parentNode->removeChild($element);
             }
 
-            $content = preg_replace("/<script.*?>.*?<\\/script>/s", "", $content);
+            $content = $doc->saveHTML();
+
+            // remove doctype, html and body tags
+            $trim_off_front = strpos($content,'<body>') + 6;
+            $trim_off_end = (strrpos($content,'</body>')) - strlen($content);
```
Comment (stentrop) [inline src/Extensions/TwigStyleScriptTagFilter.php]: camelCase

File: src/Extensions/TwigStyleScriptTagFilter.php [hunk 2]
```diff
@@ -103,35 +103,50 @@ public function filterTags($content, $containerName)
...
+
+                $element->parentNode->removeChild($element);
             }
 
-            $content = preg_replace("/<style.*?>.*?<\\/style>/s", "", $content);
+            $content = $doc->saveHTML();
+
+            // remove doctype, html and body tags
+            $trim_off_front = strpos($content,'<body>') + 6;
+            $trim_off_end = (strrpos($content,'</body>')) - strlen($content);
```
Comment (stentrop) [inline src/Extensions/TwigStyleScriptTagFilter.php]: camelCase

**Review by stentrop — CHANGES_REQUESTED** [review]

**Review by stentrop — CHANGES_REQUESTED** [review]
Vorher:
Bildschirmfoto 2020-06-19 um 09 12 12

Sobald er ein script tag entfernt:
Bildschirmfoto 2020-06-19 um 09 19 57


Achte auf das Wort Büro in der Navigation

File: src/Extensions/TwigStyleScriptTagFilter.php (hunk 1 above)
Comment (cjohannsen) [reply src/Extensions/TwigStyleScriptTagFilter.php]: done

File: src/Extensions/TwigStyleScriptTagFilter.php (hunk 2 above)
Comment (cjohannsen) [reply src/Extensions/TwigStyleScriptTagFilter.php]: done

**Review by stentrop — APPROVED** [review]

**Pull request merged**